rclcpp_lifecycle/src/lifecycle_node_interface_impl.hpp

```diff
@@ -248,6 +260,7 @@ class LifecycleNode::LifecycleNodeInterfaceImpl
   {
     (void)header;
     (void)req;
+    std::lock_guard<std::recursive_mutex> lock(state_machine_mutex_);
     if (rcl_lifecycle_state_machine_is_initialized(&state_machine_) != RCL_RET_OK) {
       throw std::runtime_error(
               "Can't get state. State machine is not initialized.");
```

Comment:
I know you didn't add this, but this seems like a serious problem.  An outside process can call this service and potentially crash this node, which doesn't seem ideal.  I would say that we shouldn't fix it here, but we should make an issue to fix it later.  Would you mind doing that?